tests/integration/pilot/circuitbreakers/main_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package circuitbreakers
+
+import (
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/proxy/envoy/v2"
+	"math"
+	"testing"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	v2Cluster "github.com/envoyproxy/go-control-plane/envoy/api/v2/cluster"
+	xdscore "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	. "github.com/onsi/gomega"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	serviceEntryConfig = `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: external-svc-mongocluster
+  namespace: test
+spec:
+  hosts:
+    - mymongodb.somedomain
+  location: MESH_EXTERNAL
+  ports:
+    - number: 80
+      name: http
+      protocol: HTTP
+  resolution: DNS
+`
+	defaultCircuitBreakerThresholds = v2Cluster.CircuitBreakers_Thresholds{
+		MaxRetries:         &wrappers.UInt32Value{Value: math.MaxUint32},
+		MaxRequests:        &wrappers.UInt32Value{Value: math.MaxUint32},
+		MaxConnections:     &wrappers.UInt32Value{Value: math.MaxUint32},
+		MaxPendingRequests: &wrappers.UInt32Value{Value: math.MaxUint32},
+	}
+
+	g galley.Instance
+	p pilot.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("circuitbreakers", m).
+		RequireEnvironment(environment.Native).
+		Setup(setup).
+		Run()
+}
+
+func setup(ctx resource.Context) error {
+	// Create the components.
+	var err error
+	g, err = galley.New(ctx, galley.Config{})
+	if err != nil {
+		return err
+	}
+	p, err = pilot.New(ctx, pilot.Config{
+		Galley: g,
+	})
+	return err
+}
+
+func TestCircuitBreakers(t *testing.T) {
```

Comment:
I don't think we need a full integration test to determine if we set some field in the CDS response. Why not just a unit test?

Reply:
You're probably right. I wanted to try the new framework, and it helped me make sure I'm not missing anything - I only found the passthrough cluster defaults overriding the circuit breakers because of this test. I can rewrite to unit tests next week

Reply:
I removed the integration test and extended a unit test accordingly